### ISSUE TITLE
TF 1.14 / keras & eager mode

### ISSUE THREAD

**Issue by tfriedel** [horovod/horovod#1229]
**Environment:**
1. Framework: TensorFlow
2. Framework version: 1.14.0-gpu
3. Horovod version: 0.16.4
4. MPI version: 4.0.0
5. CUDA version: 10.0
6. NCCL version: 2.4.7-1+cuda10.0
7. Python version: 3.6
8. OS and version: Ubuntu 18.04 / Docker
9. GCC version: 4.9
I used the docker container horovod/horovod:0.16.4-tf1.14.0-torch1.1.0-mxnet1.4.1-py3.6

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Bug report:**
Please describe errorneous behavior you're observing and steps to reproduce it.

I want to use eager mode with tensorflow.keras. I run an example with this command:
`horovodrun -np 2 -H localhost:2 python tf_keras_mnist2.py `

However I get this error:
```
[1,0]<stderr>:Traceback (most recent call last):                                                                                                                                                                    
[1,0]<stderr>:  File "tf_keras_mnist2.py", line 92, in <module>
[1,0]<stderr>:    validation_data=(x_test, y_test))                                                                                                                                                                 
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/training.py", line 780, in fit                                                                                          
[1,0]<stderr>:    steps_name='steps_per_epoch')                                                                                                                                                                     
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/training_arrays.py", line 239, in model_iteration                                                                       
[1,0]<stderr>:    callbacks._call_begin_hook(mode)
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/callbacks.py", line 262, in _call_begin_hook                                                                                   
[1,0]<stderr>:    self.on_train_begin()
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/callbacks.py", line 378, in on_train_begin                                                                                     
[1,0]<stderr>:    callback.on_train_begin(logs)
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/horovod/_keras/callbacks.py", line 30, in on_train_begin                                                                                               
[1,0]<stderr>:    self.backend.get_session().run(bcast_op)
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/backend.py", line 459, in get_session                                                                                          
[1,0]<stderr>:    session = _get_session(op_input_list)
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/backend.py", line 431, in _get_session                                                                                         
[1,0]<stderr>:    config=get_default_session_config())
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py", line 1570, in __init__                                                                                           
[1,0]<stderr>:    super(Session, self).__init__(target, graph, config=config)
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py", line 693, in __init__
[1,0]<stderr>:    self._session = tf_session.TF_NewSessionRef(self._graph._c_graph, opts)
[1,0]<stderr>:tensorflow.python.framework.errors_impl.InvalidArgumentError: Invalid device ordinal value (1). Valid range is [0, 0].
[1,0]<stderr>:  while setting up XLA_GPU_JIT device number 1
[1,1]<stderr>:Traceback (most recent call last):
[1,1]<stderr>:  File "tf_keras_mnist2.py", line 92, in <module>
[1,1]<stderr>:    validation_data=(x_test, y_test))
[1,1]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/training.py", line 780, in fit
[1,1]<stderr>:    steps_name='steps_per_epoch')
[1,1]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/training_arrays.py", line 239, in model_iteration
[1,1]<stderr>:    callbacks._call_begin_hook(mode)
[1,1]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/callbacks.py", line 262, in _call_begin_hook
[1,1]<stderr>:    self.on_train_begin()
[1,1]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/callbacks.py", line 378, in on_train_begin
[1,1]<stderr>:    callback.on_train_begin(logs)
[1,1]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/horovod/_keras/callbacks.py", line 30, in on_train_begin
[1,1]<stderr>:    self.backend.get_session().run(bcast_op)
[1,1]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/backend.py", line 459, in get_session
[1,1]<stderr>:    session = _get_session(op_input_list)
[1,1]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/backend.py", line 431, in _get_session
[1,1]<stderr>:    config=get_default_session_config())
[1,1]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py", line 1570, in __init__
[1,1]<stderr>:    super(Session, self).__init__(target, graph, config=config)
[1,1]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py", line 693, in __init__
[1,1]<stderr>:    self._session = tf_session.TF_NewSessionRef(self._graph._c_graph, opts)
[1,1]<stderr>:tensorflow.python.framework.errors_impl.InvalidArgumentError: device CUDA:0 not supported by XLA service
[1,1]<stderr>:  while setting up XLA_GPU_JIT device number 0
```
I used a modified example which I attached. 

[tf_keras_mnist2.py.txt](https://github.com/horovod/horovod/files/3407017/tf_keras_mnist2.py.txt)

Basically I changed this code at the top:
```py
from tensorflow import keras
from tensorflow.keras.datasets import mnist
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Dropout, Flatten
from tensorflow.keras.layers import Conv2D, MaxPooling2D
from tensorflow.keras import backend as K
import math
import tensorflow as tf
import horovod.tensorflow.keras as hvd

# Horovod: initialize Horovod.
hvd.init()

# Horovod: pin GPU to be used to process local rank (one GPU per process)
tf_config = tf.ConfigProto()
tf_config.gpu_options.visible_device_list = str(hvd.local_rank())
tf_config.gpu_options.allow_growth = True
tf.enable_eager_execution(config=tf_config)
```

**Comment by alsrgv** [comment]
@tfriedel, I can reproduce the issue. We're planning to enable eager execution + Keras in TensorFlow 2.0 integration.

**Comment by alsrgv** [comment]
@tfriedel, you can try installing Horovod from `tf2_keras` branch (PR #1238) and try it out.
